src/twisted/internet/address.py

```diff
@@ -117,66 +92,35 @@ class HostnameAddress(FancyEqMixin, object):
     @ivar port: An integer representing the port number.
     @type port: L{int}
     """
-    compareAttributes = ('hostname', 'port')
-
-    def __init__(self, hostname, port):
-        self.hostname = hostname
-        self.port = port
 
-
-    def __repr__(self):
-        return '%s(%s, %d)' % (
-            self.__class__.__name__, self.hostname, self.port)
-
-
-    def __hash__(self):
-        return hash((self.hostname, self.port))
+    hostname = attr.ib()
+    port = attr.ib()
 
 
 
+@attr.s(hash=False, repr=False, cmp=False)
 @implementer(IAddress)
-class UNIXAddress(FancyEqMixin, object):
+class UNIXAddress(object):
     """
     Object representing a UNIX socket endpoint.
 
     @ivar name: The filename associated with this socket.
     @type name: C{bytes}
     """
 
-    compareAttributes = ('name', )
-
-    def __init__(self, name, _bwHack = None):
-        self.name = name
-        if _bwHack is not None:
-            warnings.warn("twisted.internet.address.UNIXAddress._bwHack is deprecated since Twisted 11.0",
-                    DeprecationWarning, stacklevel=2)
-
-
-    @property
-    def name(self):
-        return self._name
-
-
-    @name.setter
-    def name(self, name):
-        """
-        On UNIX, paths are always bytes. However, as paths are L{unicode} on
-        Python 3, and L{UNIXAddress} technically takes a file path, we convert
-        it to bytes to maintain compatibility with C{os.path} on Python 3.
-        """
-        if name is not None:
-            self._name = _asFilesystemBytes(name)
-        else:
-            self._name = None
-
+    name = attr.ib(converter=lambda x: _asFilesystemBytes(x)
```

Comment:
You can simplify this with http://www.attrs.org/en/stable/api.html#attr.converters.optional